File: zad4/src/main/java/it/zad/zad4/Controllers/ApplicationController.java
package it.zad.zad4.Controllers;

import it.zad.zad4.accountBase.Account;
import it.zad.zad4.accountBase.AccountRepository;
import jakarta.servlet.http.HttpSession;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.ui.Model;

@Controller
public class ApplicationController {
    AccountRepository repository = new AccountRepository();

    @GetMapping("/")
    public String firstLog() {
        return "redirect:/application";
    }

    @GetMapping("/application")
    public String applicationPage(HttpSession session, Model model) {
        if (session.getAttribute("LoggedIn") == null) {
            return "application";
        } else {
            String role = session.getAttribute("role").toString().toLowerCase();
            Object scrapedData = model.asMap().get("scrapedData");
            if (scrapedData != null)
                model.addAttribute("scrapedData", scrapedData);

            return role;
        }
    }


    @PostMapping("/application")
    public String logIn(@RequestParam String username, @RequestParam String password, Model model, HttpSession session) {
        Account user = repository.checkAccount(username, password);

        if (user == null) {
            model.addAttribute("error", "LOGIN UNSUCCESSFUL");
            return "application";
        }

        if (user.getRole().equals("DATE")) {
            session.setAttribute("username", username);
            session.setAttribute("role", user.getRole());


            model.addAttribute("username", username);
            model.addAttribute("role",  user.getRole());

            session.setAttribute("LoggedIn", "true");

            return "date";
        }
        else if(user.getRole().equals("HANGMAN")){
            model.addAttribute("username", username);
            model.addAttribute("role", user.getRole());

            session.setAttribute("LoggedIn", "true");

            session.setAttribute("username", username);
            session.setAttribute("role", user.getRole());

            return "hangman";
        }

        return "application";
    }
}

File: zad5/Controllers/ApplicationController.cs
using Microsoft.AspNetCore.Mvc;
using zad5.Models.scaperModel;

namespace zad5.Controllers
{
    public class ApplicationController : Controller {
        private AccountRepository repository = new AccountRepository();

        [HttpGet("/")]
        public IActionResult FirstLog(){
            return RedirectToAction("ApplicationPage");
        }

        
        [HttpGet("/application")]
        public IActionResult ApplicationPage(){
             if(Request.Cookies.ContainsKey("role")){
                var model = new scaperModel {name=Request.Cookies["username"], role=Request.Cookies["role"], scrapedData=""};

                if(TempData["scrappedData"]!=null){
                    model.scrapedData = TempData["scrappedData"].ToString();
                }
                

                return View(Request.Cookies["role"].ToLower(), model);
             }
                
            return View("application");
        }

        [HttpPost("/application")]
        public IActionResult LogIn(string username, string password) {

            if(Request.Cookies.ContainsKey("role")){
                var model = new scaperModel {name=Request.Cookies["username"], role=Request.Cookies["role"], scrapedData=""};

                return View(Request.Cookies["role"].ToLower(), model);
            }

            Account user = repository.CheckAccount(username, password);

            if (user == null)
                return View("application");
    

            if (user.getRole() == "DATE"){
                Response.Cookies.Append("username", user.getName());
                Response.Cookies.Append("role", user.getRole());

                var model = new scaperModel {name=user.getName(), role=user.getRole(), scrapedData=""};

                return View("date", model);
            }

            else if (user.getRole() == "HANGMAN"){
                 Response.Cookies.Append("username", user.getName());
                Response.Cookies.Append("role", user.getRole());

                var model = new scaperModel {name=user.getName(), role=user.getRole(), scrapedData=""};

                return View("hangman", model);
            }

            return View("application");
        }

       
     }
}
